presto-testing/src/main/java/io/prestosql/testing/datatype/DataType.java

```diff
@@ -197,11 +215,16 @@ public static String binaryLiteral(byte[] value)
         return "X'" + base16().encode(value) + "'";
     }
 
-    private static <T> DataType<T> dataType(String insertType, Type prestoResultType)
+    public static <T> DataType<T> dataType(String insertType, Type prestoResultType)
```

Comment:
It was intentionally private because of its implicit use of Object::toString.

if someone wants this semantics, they should choose them explicitly, using the `dataType(String insertType, Type prestoResultType, Function<T, String> toLiteral)` overload.